### ISSUE TITLE
Create Spatial Batchnorm layer

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#6]
## Introduction
The spatial batchnorm just calls the normal batchnorm during forward/backward propagation, the difference is that we need to do some reshape/permutes on the inputs.

## Task
* Create a new Layer class
* Change the Layer container
* Create a test to validate forward/backward propagation

## References
* [Book AI](https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/batch_norm_layer.html)
* [Some hints on reddit](https://www.reddit.com/r/cs231n/comments/443y2g/hints_for_a2/)
* [Blog explanation](https://standardfrancis.wordpress.com/2015/04/16/batch-normalization/)
* [Quora](https://www.quora.com/How-do-I-apply-Batch-Normalization-to-the-convolutional-layer-of-a-CNN)
* [Paper](http://jmlr.org/proceedings/papers/v37/ioffe15.pdf)
* [Python code cs231n](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/layers.py#L694)

**Comment by leonardoaraujosantos** [comment]
All tests seems to pass, closing for now

**Comment by leonardoaraujosantos** [comment]
Hint: The std function from numpy does not work exactly like matlab
http://stackoverflow.com/questions/27600207/why-does-numpy-std-give-a-different-result-to-matlab-std
